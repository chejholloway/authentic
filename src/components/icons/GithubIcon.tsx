import React from "react";
import github from "../../assets/images/github-full-svgrepo-com.svg";
/**
 * Renders the GitHub icon as an image.
 *
 * This component displays the GitHub logo as an image. You can provide an optional
 * CSS class name to style the icon as needed.
 *
 * @component
 * @example
 * ```tsx
 * <GithubIcon className="text-gray-600" />
 * ```
 * @defaultprops
 * @param {object} props - The component props.
 * @param {string} [props.className] - An optional CSS class name to apply custom styling.
 */
interface GithubIconProps {
  className?: string;
}


const GithubIcon: React.FC<GithubIconProps> = ({ className }) => {
  return (
    <img
      className={`h-10 object-cover ${className}`}
      src={github}
      alt="Github Text Logo"
    />
  );
};

export default GithubIcon;
