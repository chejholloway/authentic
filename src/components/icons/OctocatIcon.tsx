import React from "react";
import octocat from "../../assets/images/github-octocat-svgrepo-com.svg";
/**
 * Renders the Octocat icon as an image.
 *
 * This component displays the Octocat logo as an image. You can provide an optional
 * CSS class name to style the icon as needed.
 *
 * @component
 * @example
 * ```tsx
 * <OctocatIcon className="text-gray-600" />
 * ```
 * @defaultprops
 * @param {object} props - The component props.
 * @param {string} [props.className] - An optional CSS class name to apply custom styling.
 */
interface OctocatIconProps {
  className?: string;
}

const OctocatIcon: React.FC<OctocatIconProps> = ({ className }) => {
  return (
    <img
      className={`h-10 object-cover ${className}`}
      src={octocat}
      alt="Github Octocat"
    />
  );
};

export default OctocatIcon;
