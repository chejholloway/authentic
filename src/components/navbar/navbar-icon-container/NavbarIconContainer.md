# NavbarIconContainer Component

Container component for rendering GitHub icons in the navbar.

This component serves as a container for rendering the GitHub icons (GitHub logo and Octocat)
in the navbar. It provides a link to the home page using `react-router-dom`.

## Component Details

- **Type:** React Functional Component
- **Props:**

  None

## Usage Example

```tsx
import React from 'react';
import { Link } from 'react-router-dom';
import GithubIcon from '../../icons/GithubIcon';
import OctocatIcon from '../../icons/OctocatIcon';

const NavbarIconContainer: React.FC = () => {
  return (
    <div className='flex items-center'>
      <Link to='/' className='cursor-pointer'>
        <div className='flex justify-center'>
          <GithubIcon />
          <OctocatIcon />
        </div>
      </Link>
    </div>
  );
};

export default NavbarIconContainer;
