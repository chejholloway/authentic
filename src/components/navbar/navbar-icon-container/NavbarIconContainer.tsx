import React from 'react';
import { Link } from 'react-router-dom';
import GithubIcon from '../../icons/GithubIcon';
import OctocatIcon from '../../icons/OctocatIcon';

/**
 * Container component for rendering GitHub icons in the navbar.
 *
 * This component serves as a container for rendering the GitHub icons (GitHub logo and Octocat)
 * in the navbar. It provides a link to the home page using `react-router-dom`.
 *
 * @component
 * @example
 * ```tsx
 * <NavbarIconContainer />
 * ```
 */
const NavbarIconContainer: React.FC = () => {
  return (
    <div className='flex items-center'>
      <Link to='/' className='cursor-pointer'>
        <div className='flex justify-center'>
          <GithubIcon />
          <OctocatIcon />
        </div>
      </Link>
    </div>
  );
};

export default NavbarIconContainer;
