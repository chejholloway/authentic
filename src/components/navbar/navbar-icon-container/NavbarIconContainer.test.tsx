import { render } from '@testing-library/react';
import React from 'react';
import { Link } from "react-router-dom";
import GithubIcon from "../../icons/GithubIcon";
import OctocatIcon from "../../icons/OctocatIcon";
import NavbarIconContainer from "./NavbarIconContainer";

jest.mock("react-router-dom");
jest.mock("../../icons/GithubIcon");
jest.mock("../../icons/OctocatIcon");

describe('<NavbarIconContainer>', () => {
  it('should render component', () => {
    const { container } = render(<NavbarIconContainer />);

    expect(container).toMatchSnapshot();
  });
});
