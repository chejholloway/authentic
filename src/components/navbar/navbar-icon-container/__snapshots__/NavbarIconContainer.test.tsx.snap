// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarIconContainer> should render component 1`] = `
<div>
  <div
    class="flex items-center"
  />
</div>
`;
